DDD-13.md

```diff
@@ -0,0 +1,529 @@
+
+Then the `debezium-schema-generator` could be modified to support the new format and how to retrieve the components.
+
+## How to expose descriptors
+
+Since the main user of these descriptors, as of now, will be the Debezium Platform, they cannot be obtained from a running Debezium server instance,
+as the information contained in the description needs to be used during the creation of the Debezium Platform components (Source, Destination, and Transformation).
+
+This requires the descriptors to be available to the platform conductor before running the Debezium server instance.
+
+The figure below shows the high-level architecture.
+
+![Debezium Platform - Descriptor service](DDD-13/platform-descriptor.png)
+
+In the long term, we will leverage [Image Volumes](https://kubernetes.io/docs/tasks/configure-pod-container/image-volumes/), which are moving to beta in Kubernetes 1.33. For the near future, the same can be achieved using just persistent volumes with little variation that I'll show later.
+
+The idea is that the `conductor` pod will have two images mounted as volumes: Debezium server and an image provided by the user with custom libs.
+
+Once mounted in the `conductor` pod, the new `descriptor service`, during startup, will be in charge of reading the descriptors and exposing them through an API in the form of `json`.
+
+With this approach, we maintain our Debezium server distribution as immutable while giving users the possibility to extend it by providing an image.
+
+### Alternative to Image volumes
+
+Until Image Volumes become officially available, we can leverage [init containers](http://kubernetes.io/docs/concepts/workloads/pods/init-containers/).
+Both `debezium-server` and user `custom-image` will be used as init containers for the `conductor` pod.
+A persistent volume is required to copy the descriptors from init container images and then share this volume with the regular container.
+
+## Proposed changes
+
+1. Modify the `debezium-schema-generator` to generate descriptors with the new format.
+2. Modify the `debezium-schema-generator` to generate descriptors for other components (transforms, predicates, etc.).
+   1. If we go for the KIP to have a common interface, until it is approved, we can just look for the different interfaces.
+3. Generate descriptors on Debezium Server distribution:
+   1. Add `debezium-schema-generator` plugin to `debezium-server-dist`
+   2. Modify the assembly to copy generated files to the distribution package
+4. Implement the descriptor service
+5. Support volumes in the Debezium Platform Helm chart
+
+> **_Note:_** Point 2 can be postponed to the end so that we can go thought the whole pieces having only the connector descriptors and then add the others. 
+
+## Open points
+
+* Contribute to adding deprecation/alias support to `ConfigKey`?
```

Comment:
I'd definitely try to contribute as it is something that the whole community can befenfit from but don't rely on it at east now as the lead time in Kafka can be quite slow.